pyiron_base/generic/filedata.py

```diff
@@ -0,0 +1,130 @@
+"""Generic File Object."""
+
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import json
+import os
+import pandas
+
+from pyiron_base import ImportAlarm, FileHDFio
+
+__author__ = "Niklas Siemer"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Niklas Siemer"
+__email__ = "siemer@mpie.de"
+__status__ = "development"
+__date__ = "Feb 02, 2021"
+
+
+_has_imported = {}
+import_alarm = ImportAlarm()
+_not_imported = ''
+try:
+    from PIL import Image
+    _has_imported['PIL'] = True
```

Comment:
Actually, the warning is only given once, if I am not mistaken. Thus, I should really _use_ the decorator for `load_file`. The reason for the lengthy loop over `_not_imported` was due to different packages I tried to use and this might easily taken out. The `_has_imported` dictionary exists for the same reason. I might just use 
`_has_imported_pil = True` here.
I need to know which (optional) packages are installed to allow for a working solution with reduced functionality without additional try/except later on.

Reply:
I'm not opposed to the solution for now, but if it comes up again, we should probably see if we can move that logic to `ImportAlert`.